sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -0,0 +1,27 @@
+import { Body, Controller, Post, Param } from "@nestjs/common";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty } from "../../auth/decorators";
+import { CreateEducationProgramOfferingDto } from "./models/create-education-program-offering.dto";
+import { EducationProgramOfferingService } from "../../services";
+
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("institution/offering")
+export class EducationProgramOfferingController {
+  constructor(
+    private readonly programOfferingService: EducationProgramOfferingService,
+  ) {}
+
+  @Post("location/:locationId/education-program/:programId")
```

Comment:
I am not sure about the URI, if we should maybe pass location and program id in the payload, as we talked or keep in the URI.
@ann-aot, do you have some opinion or suggestions about one way or another or to keep as is the URI?

Reply:
I think there is no need to send `locationId` , it can be accessed from the offering object

Reply:
this is `create` right, In my opinion, it should be passed as a payload

Reply:
My thoughts was, relationship between locationId and ProgramId with the offering was tightly coupled, SO thought of passing it in the Param. So in the next sprints when we want list the offering for a particular program. It would be
GET : <URL>offering/location/:locationId/education-program/:programId

Reply:
I got the idea from the GET 👍